kong/db/dao/init.lua

```diff
@@ -1011,7 +1010,9 @@ function DAO:select(pk_or_entity, options)
   end
 
   local err
-  row, err, err_t = self:row_to_entity(row, options)
+  local row_to_entity_options = utils.cycle_aware_deep_copy(options or {})
+  row_to_entity_options.expand_shorthands = true
+  row, err, err_t = self:row_to_entity(row, row_to_entity_options)
```

Comment:
Do we really want to expand the shorthands in all circumstances and all cases here? It seems a bit overkill for something that we only need in the Admin API.

Anyway, if we always want to do this, a possible way to do this without the intermediate table copies is by passing a third parameter to row_to_entity:
```suggestion
  row, err, err_t = self:row_to_entity(row, options, true)
```